generators/liquibase/generator.ts

```diff
@@ -134,6 +134,15 @@ export default class LiquibaseGenerator extends BaseEntityChangesGenerator {
           prepareFieldForLiquibase(entity, field);
         }
       },
+      validateConsistencyOfField({ entity, field }) {
+        if (!this.skipChecks) {
+          if (field.columnRequired && field.fieldHasAnyDefaultValue) {
+            throw new Error(
+              `The field ${field.fieldName} in entity ${entity.name} has both columnRequired and a defaultValue, this can lead to unexpected behaviors.`,
+            );
+          }
+        }
```

Comment:
The test should be done in new fields.
```suggestion
        if (field.columnRequired && !field.fieldHasAnyDefaultValue) {
          this.handleCheckFailure(
              `The new required field ${field.fieldName} in entity ${entity.name} doesn't have a defaultValue, this can lead to unexpected behaviors.`,
          );
        }
```